lib/recommendation.js

```diff
@@ -0,0 +1,61 @@
+var XMLHttpRequest = Cc['@mozilla.org/xmlextras/xmlhttprequest;1'].createInstance();
```

Comment:
Ah, you also need to define `Cc` before you use it. Have a look at other files in this repo to see how.